### PR TITLE
feat: Add support for float in generate_series

### LINKED ISSUE #1158: Float support for `generate_series`
## Summary

E.g.
```
SELECT * FROM generate_series(1.1, 4, 1.3);
 generate_series
-----------------
             1.1
             2.4
             3.7
(3 rows)
```

### PR DESCRIPTION
**Pull request by adhish20**
Fixes #1158

### CODE REVIEW

**Review by vrongmeal — CHANGES_REQUESTED** [review]
So we have a lot of repetition here. It'll be harder to extend this for numerics and timestamps this way later on.

We could have a trait defining schema, primitive type and a function that generates the batch from the `Vec`.

```rust
trait GenerateSeriesType {
    type PrimType: ...; // can be i64, f64 etc.
    const ArrowType: DataType; // for schema

    /// Collect a series batch into an arrow array with a compatible type to `Self::ArrowType`.
    fn collect_array(batch: Vec<Self::PrimType>) -> Arc<dyn Array>;
}
```

And hence we can have `GenerateSeriesPartition<T: GenerateSeriesType>`. A similar thing is done in the `decimal` crate.

File: crates/sqlbuiltins/src/functions.rs [hunk 1]
```diff
@@ -867,36 +895,61 @@ impl TableFunc for GenerateSeries {
...
+                    && is_scalar_float(&stop)
+                    && is_scalar_float(&step)
+                {
+                    create_straming_table_float(
+                        f64_from_scalar(start)?,
+                        f64_from_scalar(stop)?,
+                        f64_from_scalar(step)?,
+                    )
+                } else {
+                    return Err(BuiltinError::Static("Unsupported param types"));
```
[the remark this answers is not in the document]
Comment (adhish20) [reply crates/sqlbuiltins/src/functions.rs]: Added appropriate error message

**Review by vrongmeal — APPROVED** [review]
This looks good to go 🚀

File: testdata/sqllogictests/functions/generate_series.slt [hunk 2]
```diff
@@ -61,3 +61,29 @@ query I
 select sum(a) from generate_series(1, 200000) as t(a);
 ----
 20000100000
+
+query R
+select * from generate_series(1.1, 4, 1.3);
+----
+1.1
+2.4000000000000005
+3.7
```
Comment (scsmithr) [inline testdata/sqllogictests/functions/generate_series.slt:70]: Nice, I was expecting that a first pass might only have _just_ floats or _just_ ints for args. Cool that they can be mixed.

**Review by scsmithr — APPROVED** [review]
Ship it!

**Pull request merged**